### ISSUE TITLE
Allow myst being used as a crystal shard 

### ISSUE THREAD

**Issue by Jens0512** [myst-lang/myst#130]
Myst has uses as a crystal shard. 

I gave this a test and after adding Myst as a dependency to the `shard.yml` file in my crystal project,
requiring myst (`require "myst"`) always exited with the message `No source file given.`.

The reason is that in `myst.cr` (the `"myst"` entrypoint), you find this line:
https://github.com/myst-lang/myst/blob/b4b6d192d14a2554262f0d7e8b5ab154309720c1/src/myst.cr#L7

Which runs myst like it would be run from the command line. 
So the issue is to allow myst being differently than the command line. Which would require some refactoring.

This is where the program exits in the `cli.cr` file:
https://github.com/myst-lang/myst/blob/b4b6d192d14a2554262f0d7e8b5ab154309720c1/src/myst/cli.cr#L59-L62
Not that it matters very much, as the cli should only be run when used in the command line.
## 
My idea is that we can use Myst in crystal like 
```crystal
require "myst"

Myst::VM.run("IO.puts(:Hello)").output.to_s # => "Hello\n"
```

Where `VM` (just an example name), is a convenience class so everyone won't have to bother with the parser and interpreter. 

For this to work, we would need to do something about the cli being run when requiring myst, and of course the most important part of not hindering any of myst's other (and more important) goals.

**Comment by faultyserver** [comment]
My first thought on how to do this is to make the main `myst.cr` file only require all of the project files (everything under `src/myst/`). Then, the `myst` executable can be built from a second file like `src/myst_cli.cr`, which will `require "myst"`, then call `Cli.run`, like the main file currently does.

**Comment by Jens0512** [comment]
I'll get a pull request with a `myst_cli` file and the `VM` class ready

**Comment by faultyserver** [comment]
This was implemented in #134 .
